### ISSUE TITLE
selfhosted: if statement with binary literal produces parse error

### ISSUE THREAD

**Issue by mnlrsn** [SerenityOS/jakt#640]
Code snippet to reproduce:
```
function g(a: u32) {
    if ( a & 0b1 ) {
    }
}
```
Error message: 
```
 1 |
 2 | function g(a: u32) {
                        ^- Expected complete block
 3 |     if ( a & 0b1 ) {
```

**Comment by mnlrsn** [comment]
It seems to depend on the parens (`()`). This also produces an error:
```
function f() {
    let a = ( 0b10u32 & 0b11u32 )
}
```
The error goes away if the parens are removed.

**Comment by mattisboeckle** [comment]
Binary, Hexadecimal and Octal numbers were not yet supported in the selfhosted compiler.
Should be fixed by #642

**Comment by mnlrsn** [comment]
Fixed
